src/Xamarin.Android.Build.Tasks/Tests/Xamarin.Android.Build.Tests/BuildTest.cs

```diff
@@ -66,6 +66,23 @@ public void BuildBasicApplicationReleaseProfiledAotWithoutDefaultProfile ()
 			}
 		}
 
+		[Test, Category ("UsesDevice")]
+		public void BuildBasicApplicationAndAotProfileIt ()
+		{
+			if (!HasDevices)
+				Assert.Ignore ("Skipping test. No devices available.");
+
+			var proj = new XamarinAndroidApplicationProject ();
+			var projDirectory = Path.Combine ("temp", TestName);
+			using (var b = CreateApkBuilder (projDirectory)) {
+				Assert.IsTrue (b.RunTarget (proj, "BuildAndStartAotProfiling"), "Run of BuildAndStartAotProfiling should have succeeded.");
```

Comment:
Looks like something in this test method [is failing](https://devdiv.visualstudio.com/DevDiv/_build/results?buildId=3276974&view=ms.vss-test-web.build-test-results-tab&runId=9973166&resultId=100000&paneView=attachments):

```
/Library/Frameworks/Mono.framework/External/xbuild/Xamarin/Android/Xamarin.Android.Common.Debugging.targets(612,2): error : Parameter name: source [/Users/runner/runners/2.160.1/work/1/s/bin/TestRelease/temp/BuildBasicApplicationAndAotProfileIt/UnnamedProject.csproj]
/Library/Frameworks/Mono.framework/External/xbuild/Xamarin/Android/Xamarin.Android.Common.Debugging.targets(612,2): error :   at System.Linq.Enumerable.Select[TSource,TResult] (System.Collections.Generic.IEnumerable`1[T] source, System.Func`2[T,TResult] selector) [0x0000d] in <549f46d1789f4a51aa5fa1e7d715fe0d>:0  [/Users/runner/runners/2.160.1/work/1/s/bin/TestRelease/temp/BuildBasicApplicationAndAotProfileIt/UnnamedProject.csproj]
/Library/Frameworks/Mono.framework/External/xbuild/Xamarin/Android/Xamarin.Android.Common.Debugging.targets(612,2): error :   at Xamarin.Android.Tasks.InstallPackageAssemblies.RunTask () [0x003d0] in <26f94f6d10aa4e62aabb7af2f165e6d5>:0  [/Users/runner/runners/2.160.1/work/1/s/bin/TestRelease/temp/BuildBasicApplicationAndAotProfileIt/UnnamedProject.csproj]
```

Reply:
Yay, so it runs on CI :-)

I wasn't able to reproduce this one locally, but noticed another problem introduced by removing this line from the test:

    Assert.IsTrue (b.Build (proj), "Build should have succeeded.");

And failing locally with:

    error MSB3073: The command ""/Users/rodo/android-toolchain/sdk/platform-tools/adb"  shell am start -n "UnnamedProject.UnnamedProject/"" exited with code 255.

I was able to overcome that with adding `Build` target to `BuildAndStartAotProfiling` dependencies. I expected the `Install` target to be enough to trigger the `Build` target as well, but looks like it wasn't the case and as a result we didn't have launcher activity set in the manifest? @dellis1972 @jonathanpeppers is that expected behavior?

Thanks to that I also noticed we need to have `AndroidEmbedProfilers` in cached properties to trigger rebuild.

Reply:
The `Install` target does NOT always run a `Build`, it depends on certain conditions like if you are running in an IDE. All of our using tests pass `BuildingInsideVisualStudio=true` so in this case `Install` will NOT call `Build`.

Reply:
The `Install` target only calls build if we are running from the command line. In the IDE is does NOT call `Build`. Our tests run as if they are in the IDE, so a call to Build will be needed.